atoma-service/src/handlers/mod.rs

```diff
@@ -298,19 +299,84 @@ pub fn update_stack_num_compute_units(
     estimated_total_compute_units: i64,
     total_compute_units: i64,
     endpoint: &str,
+    concurrent_requests: u64,
 ) -> Result<(), AtomaServiceError> {
     state_manager_sender
         .send(AtomaAtomaStateManagerEvent::UpdateStackNumComputeUnits {
             stack_small_id,
             total_compute_units,
             estimated_total_compute_units,
+            concurrent_requests,
         })
         .map_err(|e| AtomaServiceError::InternalError {
             message: format!("Error sending update stack num compute units event: {e}"),
             endpoint: endpoint.to_string(),
         })
 }
 
+/// Handles the concurrent requests count updates for a stack.
+///
+/// This function updates the concurrent requests count for a stack and removes the stack from the map
+/// if the count reaches 0.
+///
+/// # Arguments
+///
+/// * `concurrent_requests_per_stack` - The map of stack small ids to their concurrent requests count
+/// * `stack_small_id` - The small id of the stack to update
+///
+/// # Returns
+///
+/// Returns `Ok(())` if the update was successful, or an `AtomaServiceError` if the update failed.
+///
+/// # Instrumentation
+///
+/// This function is instrumented with info-level tracing that includes:
+/// - stack_small_id
+/// - concurrent_requests_count
+/// - endpoint
+///
+/// # Example   
+///
+/// ```rust,ignore
+/// use crate::AppState;
+///
+/// async fn handle_concurrent_requests_count_updates(state: &AppState, stack_small_id: i64) -> Result<(), AtomaServiceError> {
+///     handle_concurrent_requests_count_updates(&state.concurrent_requests_per_stack, stack_small_id).await?;
+///     Ok(())
+/// }
+/// ```
+#[instrument(level = "info", skip_all, fields(stack_small_id, endpoint))]
+pub fn handle_concurrent_requests_count_updates(
```

Comment:
maybe mention in the name that's the `decrease` handler

Reply:
what ?